### ISSUE TITLE
Cannot install python package

### ISSUE THREAD

**Issue by paddyroddy** [astro-informatics/ssht#54]
I'm just trying to install `pyssht` as normal. Locally I don't encounter this issue. I'm installing this on a cluster and wonder whether it is related to https://github.com/astro-informatics/s2let/issues/37

`uname -a`
> Linux argo 4.13.0-16-generic #19-Ubuntu SMP Wed Oct 11 18:35:14 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

`python --version`
> Python 3.9.4

`which python`
> /home/pr2/miniconda3/envs/py39/bin/python

```
Collecting pyssht
  Using cached pyssht-1.3.7.tar.gz (75 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Requirement already satisfied: numpy in /home/pr2/miniconda3/envs/py39/lib/python3.9/site-packages (from pyssht) (1.20.3)
Requirement already satisfied: scipy in /home/pr2/miniconda3/envs/py39/lib/python3.9/site-packages (from pyssht) (1.6.3)
Building wheels for collected packages: pyssht
  Building wheel for pyssht (PEP 517) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/pr2/miniconda3/envs/py39/bin/python3.9 /home/pr2/miniconda3/envs/py39/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py build_wheel /tmp/tmpgg1aqqko
       cwd: /tmp/pip-install-sw31runl/pyssht_79f942455c1d4afe80bfb3c04d9bd5f2
  Complete output (122 lines):
  Not searching for unused variables given on the command line.
  CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
    Compatibility with CMake < 2.8.12 will be removed from a future version of
    CMake.

    Update the VERSION argument <min> value or use a ...<max> suffix to tell
    CMake that the project does not need compatibility with older versions.


  -- The C compiler identification is GNU 5.5.0
  -- Detecting C compiler ABI info
  -- Detecting C compiler ABI info - done
  -- Check for working C compiler: /home/pr2/.linuxbrew/bin/cc - skipped
  -- Detecting C compile features
  -- Detecting C compile features - done
  -- Configuring done
  -- Generating done
  -- Build files have been written to: /tmp/pip-install-sw31runl/pyssht_79f942455c1d4afe80bfb3c04d9bd5f2/_cmake_test_compile/build
  -- The C compiler identification is GNU 5.5.0
  -- Detecting C compiler ABI info
  -- Detecting C compiler ABI info - done
  -- Check for working C compiler: /home/pr2/.linuxbrew/bin/cc - skipped
  -- Detecting C compile features
  -- Detecting C compile features - done
  -- Downloading conan.cmake from https://github.com/conan-io/cmake-conan
  -- Conan: checking conan executable
  -- Conan: Found program /tmp/pip-build-env-un25ohwj/overlay/bin/conan
  -- Conan: Version found Conan version 1.36.0

  -- Conan executing: /tmp/pip-build-env-un25ohwj/overlay/bin/conan install /tmp/pip-install-sw31runl/pyssht_79f942455c1d4afe80bfb3c04d9bd5f2/conanfile.txt --remote conan-center --generator cmake_find_package --generator cmake_paths --build missing --settings build_type=Release --settings compiler=gcc --settings compiler.version=5
  Configuration:
  [settings]
  arch=x86_64
  arch_build=x86_64
  build_type=Release
  compiler=gcc
  compiler.version=5
  os=Linux
  os_build=Linux
  [options]
  [build_requires]
  [env]

  Version ranges solved
      Version range '>=3.3' required by 'conanfile.txt' resolved to 'fftw/3.3.9' in local cache

  conanfile.txt: Installing package
  Requirements
      fftw/3.3.9 from 'conan-center' - Cache
  Packages
      fftw/3.3.9:983af26d74ae563784719a7d83a287a4f9b3bc26 - Cache

  Installing (downloading, building) binaries...
  fftw/3.3.9: Already installed!
  conanfile.txt: Generator cmake_paths created conan_paths.cmake
  conanfile.txt: Generator cmake_find_package created FindFFTW3.cmake
  conanfile.txt: Generator txt created conanbuildinfo.txt
  conanfile.txt: Generator cmake created conanbuildinfo.cmake
  conanfile.txt: Generated conaninfo.txt
  conanfile.txt: Generated graphinfo
  -- Conan: Using autogenerated FindFFTW3.cmake
  -- Found FFTW3: 3.3.9 (found version "3.3.9")
  -- Library fftw3 found /home/pr2/.conan/data/fftw/3.3.9/_/_/package/983af26d74ae563784719a7d83a287a4f9b3bc26/lib/libfftw3.a
  -- Found: /home/pr2/.conan/data/fftw/3.3.9/_/_/package/983af26d74ae563784719a7d83a287a4f9b3bc26/lib/libfftw3.a
  -- Library fftw3 found /home/pr2/.conan/data/fftw/3.3.9/_/_/package/983af26d74ae563784719a7d83a287a4f9b3bc26/lib/libfftw3.a
  -- Found: /home/pr2/.conan/data/fftw/3.3.9/_/_/package/983af26d74ae563784719a7d83a287a4f9b3bc26/lib/libfftw3.a
  -- Found PythonInterp: /home/pr2/miniconda3/envs/py39/bin/python3.9 (found version "3.9.4")
  -- Found PythonLibs: /home/pr2/miniconda3/envs/py39/lib/libpython3.9.so (found version "3.9.4")
  -- Found Cython: /tmp/pip-build-env-un25ohwj/overlay/bin/cython
  -- Found NumPy: /usr/include (found version "1.19.3")
  -- Performing Test Weak Link MODULE -> SHARED (gnu_ld_ignore) - Success
  _modinit_prefix:PyInit_
  -- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE)
  -- Configuring done
  CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
  Please set them or make sure they are set and tested correctly in the CMake files:
  MATH_LIBRARY
      linked by target "ssht" in directory /tmp/pip-install-sw31runl/pyssht_79f942455c1d4afe80bfb3c04d9bd5f2/src/c

  -- Generating done
  CMake Generate step failed.  Build files cannot be regenerated correctly.
    File "/tmp/pip-build-env-un25ohwj/overlay/lib/python3.9/site-packages/skbuild/setuptools_wrap.py", line 582, in setup
      env = cmkr.configure(cmake_args,
    File "/tmp/pip-build-env-un25ohwj/overlay/lib/python3.9/site-packages/skbuild/cmaker.py", line 229, in configure
      raise SKBuildError(


  --------------------------------------------------------------------------------
  -- Trying "Ninja" generator
  --------------------------------
  ---------------------------
  ----------------------
  -----------------
  ------------
  -------
  --
  --
  -------
  ------------
  -----------------
  ----------------------
  ---------------------------
  --------------------------------
  -- Trying "Ninja" generator - success
  --------------------------------------------------------------------------------

  Configuring Project
    Working directory:
      /tmp/pip-install-sw31runl/pyssht_79f942455c1d4afe80bfb3c04d9bd5f2/_skbuild/linux-x86_64-3.9/cmake-build
    Command:
      cmake /tmp/pip-install-sw31runl/pyssht_79f942455c1d4afe80bfb3c04d9bd5f2 -G Ninja -DCMAKE_INSTALL_PREFIX:PATH=/tmp/pip-install-sw31runl/pyssht_79f942455c1d4afe80bfb3c04d9bd5f2/_skbuild/linux-x86_64-3.9/cmake-install -DPYTHON_EXECUTABLE:FILEPATH=/home/pr2/miniconda3/envs/py39/bin/python3.9 -DPYTHON_VERSION_STRING:STRING=3.9.4 -DPYTHON_INCLUDE_DIR:PATH=/home/pr2/miniconda3/envs/py39/include/python3.9 -DPYTHON_LIBRARY:FILEPATH=/home/pr2/miniconda3/envs/py39/lib/libpython3.9.so -DSKBUILD:BOOL=TRUE -DCMAKE_MODULE_PATH:PATH=/tmp/pip-build-env-un25ohwj/overlay/lib/python3.9/site-packages/skbuild/resources/cmake -Dtests:BOOL=OFF -Dconan_deps=ON -DfPIC=ON -DCMAKE_BUILD_TYPE:STRING=Release

  Traceback (most recent call last):

  An error occurred while configuring with CMake.
    Command:
      cmake /tmp/pip-install-sw31runl/pyssht_79f942455c1d4afe80bfb3c04d9bd5f2 -G Ninja -DCMAKE_INSTALL_PREFIX:PATH=/tmp/pip-install-sw31runl/pyssht_79f942455c1d4afe80bfb3c04d9bd5f2/_skbuild/linux-x86_64-3.9/cmake-install -DPYTHON_EXECUTABLE:FILEPATH=/home/pr2/miniconda3/envs/py39/bin/python3.9 -DPYTHON_VERSION_STRING:STRING=3.9.4 -DPYTHON_INCLUDE_DIR:PATH=/home/pr2/miniconda3/envs/py39/include/python3.9 -DPYTHON_LIBRARY:FILEPATH=/home/pr2/miniconda3/envs/py39/lib/libpython3.9.so -DSKBUILD:BOOL=TRUE -DCMAKE_MODULE_PATH:PATH=/tmp/pip-build-env-un25ohwj/overlay/lib/python3.9/site-packages/skbuild/resources/cmake -Dtests:BOOL=OFF -Dconan_deps=ON -DfPIC=ON -DCMAKE_BUILD_TYPE:STRING=Release
    Source directory:
      /tmp/pip-install-sw31runl/pyssht_79f942455c1d4afe80bfb3c04d9bd5f2
    Working directory:
      /tmp/pip-install-sw31runl/pyssht_79f942455c1d4afe80bfb3c04d9bd5f2/_skbuild/linux-x86_64-3.9/cmake-build
  Please see CMake's output for more information.
  ----------------------------------------
  ERROR: Failed building wheel for pyssht
Failed to build pyssht
ERROR: Could not build wheels for pyssht which use PEP 517 and cannot be installed directly
```

**Comment by paddyroddy** [comment]
I downgraded to `pyssht==1.3.6` and that worked okay

**Comment by paddyroddy** [comment]
Still getting this problem, but it seems to be Linux only. Related to https://github.com/astro-informatics/s2let/issues/39. Downgrading to `1.3.6` works, but no future version does.

`uname -a`
> Linux paddy 5.11.0-7620-generic #21~1626191760~21.04~55de9c3-Ubuntu SMP Tue Jul 20 22:18:55 UTC  x86_64 x86_64 x86_64 GNU/Linux

`python --version`
> Python 3.9.6

```
Requirement already satisfied: pyssht in /home/paddy/miniconda3/envs/py39/lib/python3.9/site-packages (1.3.6)
Requirement already satisfied: numpy in /home/paddy/miniconda3/envs/py39/lib/python3.9/site-packages (from pyssht) (1.21.2)
Requirement already satisfied: scipy in /home/paddy/miniconda3/envs/py39/lib/python3.9/site-packages (from pyssht) (1.7.1)
~/project/src/s2sleplet develop ❯ pip install --upgrade pyssht                                                                 py39 09:45:34
Requirement already satisfied: pyssht in /home/paddy/miniconda3/envs/py39/lib/python3.9/site-packages (1.3.6)
Collecting pyssht
  Using cached pyssht-1.5.0.tar.gz (78 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Requirement already satisfied: numpy in /home/paddy/miniconda3/envs/py39/lib/python3.9/site-packages (from pyssht) (1.21.2)
Requirement already satisfied: scipy in /home/paddy/miniconda3/envs/py39/lib/python3.9/site-packages (from pyssht) (1.7.1)
Building wheels for collected packages: pyssht
  Building wheel for pyssht (PEP 517) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/paddy/miniconda3/envs/py39/bin/python /home/paddy/miniconda3/envs/py39/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py build_wheel /tmp/tmpocndq8ez
       cwd: /tmp/pip-install-bl1kkttg/pyssht_47ad6b99639c4c5890e415e64fdc1342
  Complete output (73 lines):
  Not searching for unused variables given on the command line.
  -- The C compiler identification is GNU 10.3.0
  -- Detecting C compiler ABI info
  -- Detecting C compiler ABI info - done
  -- Check for working C compiler: /usr/bin/cc - skipped
  -- Detecting C compile features
  -- Detecting C compile features - done
  -- Configuring done
  -- Generating done
  -- Build files have been written to: /tmp/pip-install-bl1kkttg/pyssht_47ad6b99639c4c5890e415e64fdc1342/_cmake_test_compile/build
  -- The C compiler identification is GNU 10.3.0
  -- Detecting C compiler ABI info
  -- Detecting C compiler ABI info - done
  -- Check for working C compiler: /usr/bin/cc - skipped
  -- Detecting C compile features
  -- Detecting C compile features - done
  -- Downloading conan.cmake from https://github.com/conan-io/cmake-conan
  -- Conan: checking conan executable
  -- Conan: Found program /tmp/pip-build-env-nqgl0g1y/overlay/bin/conan
  -- Conan: Version found Conan version 1.40.1

  -- Conan executing: /tmp/pip-build-env-nqgl0g1y/overlay/bin/conan install /tmp/pip-install-bl1kkttg/pyssht_47ad6b99639c4c5890e415e64fdc1342/conanfile.txt --remote conan-center --generator cmake_find_package --generator cmake_paths --build missing --settings build_type=Release --settings compiler=gcc --settings compiler.version=10
  ERROR: No remote 'conan-center' defined in remotes
  CMake Error at _skbuild/linux-x86_64-3.9/cmake-build/conan.cmake:631 (message):
    Conan install failed='1'
  Call Stack (most recent call first):
    cmake/conan_dependencies.cmake:12 (conan_cmake_install)
    CMakeLists.txt:23 (include)


  -- Configuring incomplete, errors occurred!
  See also "/tmp/pip-install-bl1kkttg/pyssht_47ad6b99639c4c5890e415e64fdc1342/_skbuild/linux-x86_64-3.9/cmake-build/CMakeFiles/CMakeOutput.log".
    File "/tmp/pip-build-env-nqgl0g1y/overlay/lib/python3.9/site-packages/skbuild/setuptools_wrap.py", line 586, in setup
      env = cmkr.configure(cmake_args,
    File "/tmp/pip-build-env-nqgl0g1y/overlay/lib/python3.9/site-packages/skbuild/cmaker.py", line 237, in configure
      raise SKBuildError(


  --------------------------------------------------------------------------------
  -- Trying "Ninja" generator
  --------------------------------
  ---------------------------
  ----------------------
  -----------------
  ------------
  -------
  --
  --
  -------
  ------------
  -----------------
  ----------------------
  ---------------------------
  --------------------------------
  -- Trying "Ninja" generator - success
  --------------------------------------------------------------------------------

  Configuring Project
    Working directory:
      /tmp/pip-install-bl1kkttg/pyssht_47ad6b99639c4c5890e415e64fdc1342/_skbuild/linux-x86_64-3.9/cmake-build
    Command:
      cmake /tmp/pip-install-bl1kkttg/pyssht_47ad6b99639c4c5890e415e64fdc1342 -G Ninja -DCMAKE_INSTALL_PREFIX:PATH=/tmp/pip-install-bl1kkttg/pyssht_47ad6b99639c4c5890e415e64fdc1342/_skbuild/linux-x86_64-3.9/cmake-install -DPYTHON_EXECUTABLE:FILEPATH=/home/paddy/miniconda3/envs/py39/bin/python -DPYTHON_VERSION_STRING:STRING=3.9.6 -DPYTHON_INCLUDE_DIR:PATH=/home/paddy/miniconda3/envs/py39/include/python3.9 -DPYTHON_LIBRARY:FILEPATH=/home/paddy/miniconda3/envs/py39/lib/libpython3.9.so -DSKBUILD:INTERNAL=TRUE -DCMAKE_MODULE_PATH:PATH=/tmp/pip-build-env-nqgl0g1y/overlay/lib/python3.9/site-packages/skbuild/resources/cmake -Dtests:BOOL=OFF -Dconan_deps=ON -DfPIC=ON -DCMAKE_BUILD_TYPE:STRING=Release

  Traceback (most recent call last):

  An error occurred while configuring with CMake.
    Command:
      cmake /tmp/pip-install-bl1kkttg/pyssht_47ad6b99639c4c5890e415e64fdc1342 -G Ninja -DCMAKE_INSTALL_PREFIX:PATH=/tmp/pip-install-bl1kkttg/pyssht_47ad6b99639c4c5890e415e64fdc1342/_skbuild/linux-x86_64-3.9/cmake-install -DPYTHON_EXECUTABLE:FILEPATH=/home/paddy/miniconda3/envs/py39/bin/python -DPYTHON_VERSION_STRING:STRING=3.9.6 -DPYTHON_INCLUDE_DIR:PATH=/home/paddy/miniconda3/envs/py39/include/python3.9 -DPYTHON_LIBRARY:FILEPATH=/home/paddy/miniconda3/envs/py39/lib/libpython3.9.so -DSKBUILD:INTERNAL=TRUE -DCMAKE_MODULE_PATH:PATH=/tmp/pip-build-env-nqgl0g1y/overlay/lib/python3.9/site-packages/skbuild/resources/cmake -Dtests:BOOL=OFF -Dconan_deps=ON -DfPIC=ON -DCMAKE_BUILD_TYPE:STRING=Release
    Source directory:
      /tmp/pip-install-bl1kkttg/pyssht_47ad6b99639c4c5890e415e64fdc1342
    Working directory:
      /tmp/pip-install-bl1kkttg/pyssht_47ad6b99639c4c5890e415e64fdc1342/_skbuild/linux-x86_64-3.9/cmake-build
  Please see CMake's output for more information.
  ----------------------------------------
  ERROR: Failed building wheel for pyssht
Failed to build pyssht
ERROR: Could not build wheels for pyssht which use PEP 517 and cannot be installed directly
```

**Comment by jasonmcewen** [comment]
@paddyroddy Is this solved by the related [s2let issue #39](https://github.com/astro-informatics/s2let/issues/39).  Can we close?

**Comment by paddyroddy** [comment]
The issues are related yes, but it's not "fixed" as discussed in that other thread
